pyomo/contrib/mindtpy/tests/test_mindtpy_global_lp_nlp.py

```diff
@@ -11,11 +11,9 @@
 from pyomo.opt import TerminationCondition
 from pyomo.contrib.mcpp import pyomo_mcpp
 
-required_solvers = ('baron', 'cplex_persistent')
+required_solvers = ('ipopt', 'cplex_persistent')
```

Comment:
Do you not need a global solver for these tests?

Reply:
I think the Pyomo servers don't have `baron` installed, so these tests are skipped previously, lowering the code coverage. For these selected models, `ipopt` is enough to find a global optimal solution.

Reply:
Baron is present on all testing platforms.  On GHA it runs in demo mode (because that is a public testing platform).  A licensed version of Baron is available through the Jenkins tests (although Jenkins was down for a couple days, it should be back up now).

Reply:
Although IPOPT is able to converge in the current version to the optimal solution, I think that this is a fragile test as we have no guarantees that with any newer version of the solvers it will keep working. I would be for reverting this back to `baron`

Reply:
OK.  @ZedongPeng, if you get the global tests updated this afternoon, we should be able to get this merged in tonight in time for tomorrow's release.

Reply:
All the problems that we are testing are small enough for the demo license of BARON to work with the NLPs. Let's revent this back to `baron` but not skip if the license is valid (as the demo version should work)

Reply:
I just updated the global test. Hope it's not too late.

Reply:
> BARON demo mode handles up to 10 constraints and variables and up to 50 nonlinear operations. 
I just added the baron license check, since several models exceed this limitation. @bernalde